community/kernel/src/main/java/org/neo4j/kernel/impl/util/CopyOnWriteHashMap.java

```diff
@@ -107,12 +107,43 @@ public synchronized void putAll( Map<? extends K, ? extends V> m )
         actual = copy;
     }
 
+    /**
+     * Replace all key-value pairs in this map with key-value pairs from the given map.
+     *
+     * @param other the replacement map.
+     */
+    public synchronized void replaceAll( Map<? extends K,? extends V> other )
```

Comment:
I am a bit concerned about this method. [java.util.Map.replaceAll(Bifunction)](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#replaceAll-java.util.function.BiFunction-) exists in java 8 and is defined differently - only existing mappings are altered, no new ones are added. This change seems to have the potential to cause problems in forward merging and maintaining the CoWHM implementation.